cl/search/views.py

```diff
@@ -174,6 +176,26 @@ def do_search(
     )
     search_summary_str = search_form.as_text(court_count, court_count_human)
 
+    # If the search query contains a single "cites" term (e.g., "cites:(123)"),
+    # calculate and append citation depth information to the result list
+    cited_cluster = None
+    cites_query_matches = re.findall(r"cites:\((\d+)\)", cd["q"])
+    if len(cites_query_matches) == 1:
+        try:
+            cited_cluster = OpinionCluster.objects.get(
+                pk=cites_query_matches[0]
+            )
+        except OpinionCluster.DoesNotExist:
+            pass
+        else:
+            for result in paged_results.object_list:
+                result[
+                    "references_count"
+                ] = get_citation_depth_for_cluster_pks(
+                    citing_cluster_pk=result["id"],
+                    cited_cluster_pk=cited_cluster.pk,
+                )
```

Comment:
This looks pretty good, but can we move it to its own function, perhaps in search_utils.py? And can we add a note, perhaps, about why we only do it for single matches?

Reply:
Fixed in b3669b4, though I'm not entirely satisfied by my refactor. Couldn't think of a good name for the helper function -- what do you think?

Reply:
It looks about right. The name can probably be simplified to something like `add_depth_counts`?

Reply:
👍